libs/permission-checker/src/Check/RunnerSyncApi/CanResolveSharedCodeAndVariables.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\PermissionChecker\Check\RunnerSyncApi;
+
+use Keboola\PermissionChecker\Exception\PermissionDeniedException;
+use Keboola\PermissionChecker\PermissionCheckInterface;
+use Keboola\PermissionChecker\StorageApiToken;
+
+class CanResolveSharedCodeAndVariables implements PermissionCheckInterface
+{
+    public function checkPermissions(StorageApiToken $token): void
+    {
+        $componentIds = [
+            'keboola.shared-code',
+            'keboola.variables',
+        ];
+
+        foreach ($componentIds as $componentId) {
+            if (!$token->hasAllowedComponent($componentId)) {
+                throw new PermissionDeniedException(sprintf(
+                    'You do not have permission to read configurations of "%s" component',
+                    $componentId
+                ));
```

Comment:
```suggestion
                throw PermissionDeniedException::missingComponent($componentId);
```

Reply:
No missing component mluvi o spousteni akci. Tady se ale nic nespousti, proto jsem to udelal pres vlastni message

Reply:
Jo, chapu. Tak by se to dalo pretextovat na obecnejsi "nepas pristup ke komponente". Ale to ted neresme, nech to jak to mas :+1: